modules.d/06dbus/module-setup.sh

```diff
@@ -1,70 +1,114 @@
-#!/bin/bash
+#!/bin/sh
+# This file is part of dracut.
+# SPDX-License-Identifier: GPL-2.0-or-later
 
-# called by dracut
+# Prerequisite check(s) for module.
 check() {
-  require_binaries dbus-daemon || return 1
 
-  return 255
-}
+    # If the binary(s) requirements are not fulfilled
+    # return 1 to not include the binary.
+    require_binaries busctl || return 1
+    require_binaries dbus-daemon || return 1
+    require_binaries dbus-send || return 1
 
-depends() {
-  echo systemd
-  return 0
+    # If the module dependency requirements are not fulfilled
+    # return 1 to not include the required module(s).
+    if ! dracut_module_included "systemd"; then
+        derror "dbus needs systemd in the initramfs."
+        return 1
+    fi
+
+    if ! dracut_module_included "systemd-sysusers"; then
```

Comment:
We cannot mandate sysusers here. Not all distros use it atm. If you want that, please add a logic like "if ! dracut_module_included "systemd-sysusers"; then add users manually"

Reply:
Even better: check how the distro injects system users...